### DISCUSSION TITLE
[FR] Mode tablette automatique sur HP Probook x360 11 G5 EE

### DISCUSSION THREAD

**Discussion by loulou64490** [sebanc/brunch#1757] [Q&A]
Bonjour à tous, j'ai une question.
Je suis élève en 2nd dans la région Occitanie et des ordinateurs portables nous ont été offerts.
J'ai installé une version modifiée de Chrome OS, tout fonctionne, sauf le mode tablette automatique.
J'aimerais créer un programme qui active automatiquement ce mode tablette, le seul problème, c'est que je ne connais pas l'API de mon ordi qui définit si le mode tablette est actif, et je n'ai aucune compétence en programmation.
Voici toutes les infos nécéssaire si vous voulez m'aider :
- référence ordi: HP Probook x360 11 G5 EE
- version Chrome OS: rammus R107
- info patch: https://github.com/sebanc/brunch
- version patch: Brunch r107 stable 20221029
Dernière info, pour activer le "tablet mode', la commande est:
`echo 1 | sudo tee /sys/bus/platform/devices/tablet_mode_switch.0/tablet_mode`
Et remplacer par 0 pour désactiver

**Comment by sebanc** [comment]
Globalement, le mode tablette est généralement mal supporté par le noyau linux.

Visiblement si tu as ce fichier: "/sys/bus/platform/devices/tablet_mode_switch.0/tablet_mode" c'est que tu as activé l'option "sysfs_tablet_mode" qui désactive l'autodétection du mode tablette. As-tu essayé sans cette option ?

Si ça ne fonctionne pas sans l'option "sysfs_tablet_mode", Je peux tenter quelque chose (pas sur que ça fonctionne) mais j'aurais besoin que tu lances une distribution linux sur une clé usb et que tu postes le résultat des 2 commandes ci-dessous:
- "sudo dmesg"
- "sudo lsmod"

**Comment by loulou64490** [comment]
Merci de ta réponse!
J'ai en premier bien sur essayer sans cette option, mais évidemment ça ne fonctionnait pas.
J'ai regardé sur Github, ils détecte le mode tablette en fonction de l'activation du trackpad, car il est désactivé automatiquement quand l'ordinateur est retourné. Voici le [lien du projet](https://github.com/FrancescoCaracciolo/OSK-Touchpad-inverse-toggle-Gnome-Ext/blob/main/extension.js).

Voici les commandes:
[sudo dmesg](https://pastebin.com/7dzSdnXM)
[sudo lsmod](https://pastebin.com/D44v3fSG)

En tout cas merci, mais ne t'embête pas trop, c'était je juste comme ça!

**Reply by sebanc** [reply]
J'ai tenté quelque chose dans la version de brunch ci-dessous si tu veux essayer:
https://github.com/sebanc/brunch-unstable/releases/tag/r109-unstable-20230205

**Reply by loulou64490** [reply]
Bon, je l'ai installé, à partir de 0, j'ai fait une config avec le kernel 6.1 qui, je remarque, est bien mieux supporté par mon ordi, sans rien activé de base à part le pilote wifi et hyperthreading, ça ne marchait pas. J'ai refait avec en plus l'option sysfs tablet mode mais ça ne fonctionnait pas. Bon, ne t'embete pas, je ne vais pas passer plus de temps sur chrome os car finalement, ça reste un peu limité, surtout que les apps androids fonctionnent mal. Merci quand même et bon courage !
